src/udp_socket.cpp

```diff
@@ -311,6 +311,16 @@ void udp_socket::on_read(error_code const& ec, udp::socket* s)
 
 		if (err == boost::asio::error::would_block || err == boost::asio::error::try_again) break;
 		on_read_impl(ep, err, bytes_transferred);
+
+		// found on iOS, socket will be disconnected when app goes backgroud. try to reopen it.
+		if (err == boost::asio::error::not_connected) 
+		{
+			ep = s->local_endpoint(err);
+			if (!err) {
```

Comment:
in case there is an error here, it's probably not safe to still re-issue setup_read(s) (because it will probably take us right back to this function with an error). I think just returning is better, it will basically disable receiving on the socket.